### PR TITLE
Unbreak appdaemon

### PR DESCRIPTION
**Pull request by freezeboy**
###### Motivation for this change

Upstream update, they have a lot of version constraints, I kept some and relaxed others, as it is python, I guess the only way to be sure everything is working correclty is with a full install, it someone has de test platform for it.

In particular, I replaced the `azure==4.0.0` with a split of `azure-*` python modules, I am not sure how safe it is as I don't use azure.

They rely on a lot of "old" releases of their dependencies

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by SuperSandro2000** [comment]
Result of `nixpkgs-review pr 104173` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package failed to build:</summary>
  <ul>
    <li>appdaemon</li>
  </ul>
</details>
<details>
  <summary>2 packages built:</summary>
  <ul>
    <li>python37Packages.sockjs</li>
    <li>python38Packages.sockjs</li>
  </ul>
</details>

**Comment by freezeboy** [comment]
Can you give more details on the problem please?
locally it looks ok on my side

**Comment by SuperSandro2000** [comment]
False negative again...

**Comment by SuperSandro2000** [comment]
Result of `nixpkgs-review pr 104173` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>3 packages built:</summary>
  <ul>
    <li>appdaemon</li>
    <li>python37Packages.sockjs</li>
    <li>python38Packages.sockjs</li>
  </ul>
</details>

File: pkgs/servers/home-assistant/appdaemon.nix [hunk 1]
```diff
@@ -4,61 +4,62 @@ let
...
+      --replace "pid==2.2.5" "pid" \
+      --replace "Jinja2==2.11.1" "Jinja2" \
+      --replace "pytz==2019.3" "pytz" \
+      --replace "aiohttp==3.6.2" "aiohttp>=3.6" \
+      --replace "iso8601==0.1.12" "iso8601>=0.1" \
+      --replace "azure==4.0.0" "azure-mgmt-compute
+    azure-mgmt-storage
+    azure-mgmt-resource
+    azure-keyvault-secrets
+    azure-storage-blob" \
```
Comment (jonringer) [inline pkgs/servers/home-assistant/appdaemon.nix:57]: we should let upstream know that the `azure` meta packages have been deprecated for more than a year, and your patch is the preferred method

**Review by jonringer — APPROVED** [review]
LGTM

Result of nixpkgs-review pr 104173 [1](https://github.com/Mic92/nixpkgs-review)<details>  <summary>3 packages built:</summary>  <ul>    <li>appdaemon</li>    <li>python37Packages.sockjs</li>    <li>python38Packages.sockjs</li>  </ul></details>

File: pkgs/servers/home-assistant/appdaemon.nix (hunk 1 above)
Comment (freezeboy) [reply pkgs/servers/home-assistant/appdaemon.nix:57]: Opened an issue there, thank you

**Pull request merged**